### PR TITLE
bugfix(definition): add lifecycle events and refine extend functions

### PR DESCRIPTION
**Pull request by marcalexiei**
## Description:
Provide some fix related to typings:
* add lifecycle events on `BaseInitOpts`
* refine types on Extend function both on `Ractive` and `Static`
* fix some typos

## Fixes the following issues:
#3348

## Is breaking:
No

### CODE REVIEW

**Comment by evs-chris** [comment]
Excellent. Do you need this as a release build soonish?

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,12 @@
   * Methods that set a single value will now return the new value as the resolution of the returned promise e.g. `r.toggle('foo')` if `foo` is falsey will return a promise that resolves to `true`. `add`/`subtract`/`set` behave similarly.
   * Decorators and custom events can now access their local context, in the case of a yielded attribute partial, with `this.getLocalContext()`.
 
+# 1.2.10, 1.3.13
```
Comment (evs-chris) [inline CHANGELOG.md:11]: Not a big deal at all, but I usually put an (unreleased) on there for versions that haven't been pushed to their release branch yet.

Comment (marcalexiei) [reply CHANGELOG.md:11]: Got it! I'll do it next time.

**Comment by marcalexiei** [comment]
At the moment we are in development phase so we can use the `1.x.x-build-*` tag, so no need to publish. Thanks!

**Pull request merged**